spec/JSCompilerSpec.hs

```diff
@@ -0,0 +1,173 @@
+module JSCompilerSpec (spec) where
+
+  import           Test.Hspec
+  import           Language.Mulang.Ast
+  import           Language.Mulang.JSCompiler
+  
+  spec :: Spec
+  spec = do
+    -- It would be much better to test this by running the JS code instead of checking the string output...
+    -- Also, I don't care about output format right now, so all \n are ignored.
+    describe "JS Compilation" $ do
+
+      it "MuNumber" $ do
+        (MuNumber 5) `shouldBeCompiledTo` "new MuNumber(5.0)"
+
+      it "MuNumber" $ do
+        (MuBool True) `shouldBeCompiledTo` "new MuBool(true)"
+        
+      it "MuString" $ do
+        (MuString "foo") `shouldBeCompiledTo` "new MuString(\"foo\")"
+
+      it "MuSymbol" $ do
+        (MuSymbol "foo") `shouldBeCompiledTo` "new MuSymbol(\"foo\")"
+
+      it "MuTuple" $ do
+        (MuTuple [MuNumber 1, MuNumber 2, MuNumber 3]) `shouldBeCompiledTo` "new MuTuple([new MuNumber(1.0), new MuNumber(2.0), new MuNumber(3.0)])"
+
+      it "MuList" $ do
+        (MuList [MuNumber 1, MuNumber 2, MuNumber 3]) `shouldBeCompiledTo` "new MuList([new MuNumber(1.0), new MuNumber(2.0), new MuNumber(3.0)])"
+
+      it "Enumeration" $ do
+        (Enumeration "E" ["A", "B", "C"]) `shouldBeCompiledTo` (
+          "var E = { A: 0, B: 1, C: 2 }; " ++
+          "var A = E['A'];" ++
+          "var B = E['B'];" ++
+          "var C = E['C'];"
+          )
+
+      it "EntryPoint" $ do
+        (EntryPoint "foo" (MuBool True)) `shouldBeCompiledTo` "function foo() { return new MuBool(true) }"
```

Comment:
OK, never mind, we should definitely no execute the entry points, since it could have many and it is resposibility of the code running system to call it :sweat_smile: